Update Google Cloud SQL doc to support IAM Authentication (#305)
pganalyze/pganalyze-docs: install/google_cloud_sql/_01_create_monitoring_user_iam.mdx

import { MonitoringUserBaseForGCP, MonitoringUserPerDatabaseHelpers } from "../../components/MonitoringUserSetupInstructions";
import VerifyConnection from "../../components/VerifyConnection";

### Create a service account for monitoring user

Cloud SQL supports IAM database authentication, which allows users to connect to the database using Google Cloud IAM credentials instead of traditional passwords.
This means you can use a service account as a monitoring user.

First, create a new service account with the ID "pganalyze", and set the following two roles:
* **Cloud SQL Client** (`roles/cloudsql.client`)
  * Provides connectivity access to Cloud SQL instances
* **Cloud SQL Instance User** (`roles/cloudsql.instanceUser`)
  * Role allowing access to a Cloud SQL instance

The email address `pganalyze@your-project-id.iam.gserviceaccount.com` will be given to this service account.

> For the database username, Cloud SQL for PostgreSQL truncates the `.gserviceaccount.com` suffix in the email,

Author: I agree with you, so this is simply FYI (I'll update this with your suggestion) but I copy&pasted things from Google's doc and they used "truncates" for this one:
https://cloud.google.com/sql/docs/postgres/add-manage-iam-users#creating-a-database-user

therefore the username for this service account becomes `pganalyze@your-project-id.iam`.
This username will be used in the later step, as well as in the last step of this tutorial.

### Configure new and existing instances for IAM database authentication

Once the service account for a monitoring user is created, configure the database to monitor for IAM database authentication.
Follow the [Cloud SQL documentation](https://cloud.google.com/sql/docs/postgres/create-edit-iam-instances) to configure.

You can skip this part if your database is already configured for IAM database authentication.

### Add a service account to a Cloud SQL instance

> Next, add a service account to a Cloud SQL instance to monitor. This is a step to create an actual Postgres user using this service account.

Reviewer: ```suggestion
Next, add a service account to a Cloud SQL instance to monitor. This creates an actual Postgres user for this service account.
```

> Following is the steps to add using the Google Cloud console. Please refer the

Reviewer: ```suggestion
The following steps add the service account using the Google Cloud console. Please refer to the
```

[Cloud SQL documentation](https://cloud.google.com/sql/docs/postgres/add-manage-iam-users#creating-a-database-user)

> for the other ways to add it.

Reviewer: ```suggestion
for other ways to add it.
```
